File: web-sources/S8-pkgs-ui-carbide/boxes/b2/B2Column.js

import { S8WebFront } from "/S8-pkgs-ui-carbide/S8WebFront.js";
import { S8Object } from "/S8-api/S8Object.js";

S8WebFront.CSS_import('/S8-pkgs-ui-carbide/boxes/b2/B2Box.css');


/**
 * 
 */
export class B2Column extends S8Object {

    /** @type {HTMLDivElement} */
    wrapperNode;
    
    /**
     * 
     */
    constructor(){
        super();
        this.wrapperNode = document.createElement("div");
        this.wrapperNode.classList.add("b2struct-column");
    }


    /**
     * 
     * @returns 
     */
    getEnvelope(){
        return this.wrapperNode;
    }


    /** @param {S8Object[]} elements */
    S8_set_elements(elements){
       S8WebFront.setS8Elements(this.wrapperNode, elements);
    }


    /**
     * 
     * @param {number} width 
     */
    S8_set_width(width){
        if(width < 0){ /* clear width */
            this.wrapperNode.style.removeProperty("width");
        }
        else if(width > 0){
            this.wrapperNode.style.width = `${width * 100}vw`;
        }
    }


    /** render */
    S8_render(){}


    /** dispose */
    S8_dispose(){}

}

File: sources/com/s8/pkgs/ui/carbide/boxes/b2/B2Column.java
package com.s8.pkgs.ui.carbide.boxes.b2;

import com.s8.api.web.S8WebFront;
import com.s8.api.web.S8WebObject;


/**
 * 
 * @author pierreconvert
 *
 */
public class B2Column extends S8WebObject {

	
	/**  */
	public final static String JS_TYPENAME = "/B2Column";
	
	
	/**
	 * 
	 * @param branch
	 */
	public B2Column(S8WebFront branch) {
		super(branch, B2Box.WEBPATH + JS_TYPENAME);
	}
	
	
	
	/**
	 * 
	 * @param name
	 */
	public void setElements(S8WebObject... elements) {
		vertex.outbound().setObjectListField("elements", elements);
	}
	
	
	public void setWidth(double width) {
		vertex.outbound().setFloat32Field("width", (float) width);
	}
	
}
